deepmd/train/trainer.py

```diff
@@ -321,15 +329,22 @@ def _build_lr(self):
 
     def _build_network(self, data):        
         self.place_holders = {}
-        data_dict = data.get_data_dict()
-        for kk in data_dict.keys():
-            if kk == 'type':
-                continue
-            prec = GLOBAL_TF_FLOAT_PRECISION
-            if data_dict[kk]['high_prec'] :
-                prec = GLOBAL_ENER_FLOAT_PRECISION
-            self.place_holders[kk] = tf.placeholder(prec, [None], name = 't_' + kk)
-            self.place_holders['find_'+kk] = tf.placeholder(tf.float32, name = 't_find_' + kk)
+        if self.is_compress :
+            for kk in ['coord', 'box']:
+                self.place_holders[kk] = tf.placeholder(GLOBAL_TF_FLOAT_PRECISION, [None], 't_' + kk)
+            for kk in ['energy', 'force', 'virial', 'atom_ener', 'atom_pref']:
+                self.place_holders[kk] = tf.placeholder(GLOBAL_TF_FLOAT_PRECISION, [None], 't_' + kk)
+                self.place_holders['find_' + kk] = tf.placeholder(tf.float32, [None], 't_' + kk)
```

Comment:
`energy`, `force`, ... is not a complete list! You should check the `data_requirement` in `deepmd/common.py` for a complete list of required data. 

The place_holder builders for the standard and compress modes should be unified.